### PR TITLE
NF: test for file/directory created during directory transfer

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
as requested by @david-allison

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/servicelayer/scopedstorage/MoveDirectoryTest.kt [hunk 1]
```diff
@@ -149,6 +147,66 @@ class MoveDirectoryTest {
         assertThat("file after failure was copied", afterFile!!.exists(), equalTo(false))
     }
 
+    @Test
+    fun adding_file_during_move_is_not_fatal() {
+        adding_during_move_helper() {
+            val new_file = File(it.directory, "new_file.txt")
+            IOUtils.writeStringToFile(new_file, "new file")
+            new_file.deleteOnExit()
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/servicelayer/scopedstorage/MoveDirectoryTest.kt:155]: nit: would be useful as a `createTransientFile()` overload

**Review by david-allison — APPROVED** [review]
Nitpick only. This looks great, thanks!

Note: I do believe this would be an error on `Compat21`

I'm technically happy with this outcome, although `DeleteEmptyDirectory` would fail, retrying the operation would succeed, which would be the obvious error correction case if it fails.

**Comment by Arthur-Milchior** [comment]
> I'm technically happy with this outcome, although DeleteEmptyDirectory would fail, retrying the operation would succeed, which would be the obvious error correction case if it fails.

I'm very sorry David, but I fear I'm starting to have trouble to understand you recently. In this case, I fear this sentence has far two many information in it.
I have no idea whether you are describing the current state of the code in master, what would be the ideal state of the code.
"retrying the operation", gramatically, I'd assume "the operation" is the last one mentioned `DeleteEmptyDirectory`, but i don't understand why you'd retry this one. So I assume it means "retrying moving the directory, as we will surely do if a first try led to errors".

**Review by mikehardy — APPROVED** [review]
looks like an interesting / useful test to me - I have to say, that Kotlin code is some real sorcery! compact, easy to understand sorcery, but slicing in the executable helper block for the code to run in between two moves and returning the whole closure as the test, nice.

**Comment by david-allison** [comment]
> > I'm technically happy with this outcome, although DeleteEmptyDirectory would fail, retrying the operation would succeed, which would be the obvious error correction case if it fails.
> 
> I'm very sorry David, but I fear I'm starting to have trouble to understand you recently. In this case, I fear this sentence has far two many information in it.
> I have no idea whether you are describing the current state of the code in master, what would be the ideal state of the code.
> "retrying the operation", gramatically, I'd assume "the operation" is the last one mentioned `DeleteEmptyDirectory`, but i don't understand why you'd retry this one. So I assume it means "retrying moving the directory, as we will surely do if a first try led to errors". 

I'll aim to be more clear. Retrying "move directory" would succeed.

**Comment by Arthur-Milchior** [comment]
> looks like an interesting / useful test to me - I have to say, that Kotlin code is some real sorcery! compact, easy to understand sorcery, but slicing in the executable helper block for the code to run in between two moves and returning the whole closure as the test, nice.

You lost me @mikehardy. I am not sure, but I suppose the closure you mention is the new Operation. However, this operation is not a test. I don't see anything that can be considered both as a closure and as a test here.

But, while doing compliments, I really appreciate looking at David's example. Because I would not have known about spy, nor thought about using it this way, without seeing other tests.

**Comment by mikehardy** [comment]
#5304

**Pull request merged**